account_invoice_fiscal_position_update/models/account_move.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2011-2020 Julius Network Solutions SARL <contact@julius.fr>
+# Copyright 2014-2020 Akretion France (http://www.akretion.com)
+# @author Mathieu Vatel <mathieu _at_ julius.fr>
+# @author Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id(self):
+        fiscal_position = self.fiscal_position_id
+        res = super()._onchange_partner_id()
+        if fiscal_position != self.fiscal_position_id:
+            self.fiscal_position_change()
+        return res
+
+    @api.onchange("fiscal_position_id")
+    def fiscal_position_change(self):
+        """Updates taxes and accounts on all invoice lines"""
+        res = {}
+        lines_without_product = []
+        fp = self.fiscal_position_id
+        for line in self.invoice_line_ids.filtered(lambda l: not l.display_type):
+            if line.product_id:
+                account = line._get_computed_account()
+                # we could use _get_computed_taxes()
+                product = line.with_company(self.company_id.id).product_id
+                if self.is_sale_document(include_receipts=True):
+                    # M2M fields don't have an option 'company_dependent=True'
+                    # so we need per-company post-filtering
+                    taxes = product.taxes_id.filtered(
+                        lambda tax: tax.company_id == self.company_id
+                    )
+                elif self.is_purchase_document(include_receipts=True):
+                    taxes = product.supplier_taxes_id.filtered(
+                        lambda tax: tax.company_id == self.company_id
+                    )
+                taxes = taxes or account.tax_ids.filtered(
+                    lambda tax: tax.company_id == self.company_id
+                )
+                if fp:
+                    taxes = fp.map_tax(taxes)
+
+                line.tax_ids = [(6, 0, taxes.ids)]
+                line._onchange_mark_recompute_taxes()
+                line.account_id = account.id
+            else:
+                lines_without_product.append(line.name)
+
+        self.with_context(check_move_validity=False)._recompute_dynamic_lines()
+        self._check_balanced()
```

Comment:
```suggestion
        self._recompute_dynamic_lines()
```
Not sure if there was a previous reasoning to separating these steps as the context is to stop it from doing the `_check_balanced` internally but then does it immeditately after, but I verified the error @kittiu found and by making this change it fixed it. The lines recalculated their taxes correctly before saving the invoice with each change of the fiscal position

Reply:
I have tested this PR locally and the solution that @hildickethan  proposes has worked well for me and I think it is correct.
Also, I think that if we pass the paramenter `recompute_tax_base_amount=True` to that method (`_recompute_dynamic_lines`), we can remove `line._onchange_mark_recompute_taxes()` added a few lines above.
@alexis-via tell me if you plan to continue with this PR, if not, let me know to open a new one with those changes in order to move forward with this migration :). Thanks.

Reply:
@ernestotejeda Feel free to superseed this PR, I'm too busy with other developments at the moment.